model/DataQuery.php

```diff
@@ -661,6 +664,15 @@ public function applyRelation($relation) {
 					$this->query->addLeftJoin($componentBaseClass,
 						"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
```

Comment:
How about the alias being the specific `$relation` up to this point? E.g.

```php
$dataquery->applyRelation(‘Images.Banner.Page’);
```

Will have a table joining the Image, Banner, and Page tables, with aliases Images, Images_Banner, and Images_Banner_Page